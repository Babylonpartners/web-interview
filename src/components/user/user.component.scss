@import 'assets/styles/_constants.scss';

.user {
  display: flex;
  align-items: center;

  &__name {
    font-size: 12rem;
    margin-left: $gutter-size--large;
    font-weight: 400;
    color: $bold-text-color;
  }

  &__avatar {
    width: 40rem;
    margin: -1.5rem; // to remove whitespace around image
  }
}
