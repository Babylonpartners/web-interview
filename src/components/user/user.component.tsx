import React, { FunctionComponent } from 'react'

import { UserType } from 'types/user.type'

import './user.component.scss'

export const User: FunctionComponent<{ user: UserType }> = ({ user }) => {
  return (
    <div className="user">
      <img className="user__avatar" src={user.avatar} alt={user.firstName} />
      <span className="user__name">{`${user.firstName} ${user.lastName}`}</span>
    </div>
  )
}
