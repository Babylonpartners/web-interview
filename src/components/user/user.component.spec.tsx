import React from 'react'
import renderer from 'react-test-renderer'

import { User } from './user.component'

describe('User', () => {
  const testUser = {
    id: 2,
    avatar: '',
    firstName: 'Kenny',
    lastName: 'McCormick',
  }

  it('should match snapshot', () => {
    const component = renderer.create(<User user={testUser} />)

    expect(component.toJSON()).toMatchSnapshot()
  })
})
