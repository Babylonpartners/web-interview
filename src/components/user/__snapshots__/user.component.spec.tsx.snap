// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should match snapshot 1`] = `
<div
  className="user"
>
  <img
    alt="Kenny"
    className="user__avatar"
    src=""
  />
  <span
    className="user__name"
  >
    Kenny McCormick
  </span>
</div>
`;
